deparse.c

```diff
@@ -1058,23 +1112,33 @@ sqlite_foreign_expr_walker(Node *node,
 			}
 			break;
 		case T_DistinctExpr:
-			/* IS DISTINCT FROM */
-			return false;
+			{
+				/* IS DISTINCT FROM */
+				elog(DEBUG2, "sqlite_fdw : %s T_DistinctExpr", __func__);
+				return false;
+			}
 		default:
+			{
 
-			/*
-			 * If it's anything else, assume it's unsafe.  This list can be
-			 * expanded later, but don't forget to add deparse support below.
-			 */
-			return false;
+				elog(DEBUG1, "sqlite_fdw : %s other", __func__);
+				/*
+				 * If it's anything else, assume it's unsafe.  This list can be
+				 * expanded later, but don't forget to add deparse support below.
+				 */
+				return false;
+			}
 	}
 
 	/*
-	 * If result type of given expression is not built-in, it can't be sent to
+	 * If result type of given expression is not built-in or PostGIS, it can't be sent to
 	 * remote because it might have incompatible semantics on remote side.
 	 */
-	if (check_type && !sqlite_is_builtin(exprType(node)))
-		return false;
+	if (check_type)
+	{
+		Oid typeOid = exprType(node);
+		if (!sqlite_is_builtin(typeOid) && listed_datatype_oid(typeOid, -1, postGisSQLiteCompatibleTypes))
```

Comment:
@mkgrgis I think this condition is not correct.  
This condition will not push down if data type is PostGIS data type.  
Example, `geometry` type is not built-in and it is PostGIS type, so it will be not pushed down (return false).

We should change condition to be like this.
```
if (!(sqlite_is_builtin(typeOid) || listed_datatype_oid(typeOid, -1, postGisSQLiteCompatibleTypes)))
```

Reply:
This is very good catch, @son-phamngoc -san! Fixed. Please verify.

Reply:
Thanks @mkgrgis . I confirmed. Please help me to resolve this comment.